packages/back-end/src/models/SegmentModel.ts

```diff
@@ -47,4 +47,10 @@ export class SegmentModel extends BaseClass {
   ): Promise<SegmentInterface[]> {
     return await this._find({ datasource: datasourceId });
   }
+
```

Comment:
We should add a bunch of validation to this model (customValidation hook). If type is FACT, make sure fact table isn't empty, etc.